server/service/endpoint_utils.go

```diff
@@ -211,7 +232,7 @@ func makeDecoder(iface interface{}) kithttp.DecodeRequestFunc {
 			}
 
 			_, jsonExpected := f.Tag.Lookup("json")
-			if jsonExpected && nilBody {
+			if jsonExpected && nilBody && !isBodyDecoder {
```

Comment:
Is this check actually required? Wouldn't something that implements `bodyDecoder` still expect a body?